test/spec/Feature/Query/CustomMediaSpec.hs

```diff
@@ -0,0 +1,221 @@
+      liftIO $ do
+        simpleBody r `shouldBe` readFixtureFile "lines.twkb"
+        simpleHeaders r `shouldContain` [("Content-Type", "application/vnd.twkb")]
+
+    it "fails if doesn't have an aggregate defined" $ do
+      request methodGet "/rpc/get_lines"
+          (acceptHdrs "application/octet-stream") ""
+        `shouldRespondWith`
+          [json| {"code":"PGRST107","details":null,"hint":null,"message":"None of these media types are available: application/octet-stream"} |]
+          { matchStatus = 415 }
+
+    it "works if there's an anyelement aggregate defined" $ do
+      request methodGet "/rpc/get_lines" (acceptHdrs "application/vnd.geo2+json") ""
+        `shouldRespondWith`
+        "\SOH{\"type\": \"FeatureCollection\", \"hello\": \"world\"}"
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "application/vnd.geo2+json"]
+        }
+
+  context "overriding" $ do
+    it "will override the application/json handler for a single table" $
+      request methodGet "/ov_json" (acceptHdrs "application/json") ""
+        `shouldRespondWith`
+        [json| {"overridden": "true"} |]
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "application/json; charset=utf-8"]
+        }
+
+    it "will override the application/geo+json handler for a single table" $
+      request methodGet "/lines?id=eq.1" (acceptHdrs "application/geo+json") ""
+        `shouldRespondWith`
+        "\SOH{\"crs\": {\"type\": \"name\", \"properties\": {\"name\": \"EPSG:4326\"}}, \"type\": \"FeatureCollection\", \"features\": [{\"type\": \"Feature\", \"geometry\": {\"type\": \"LineString\", \"coordinates\": [[1, 1], [5, 5]]}, \"properties\": {\"id\": 1, \"name\": \"line-1\"}}]}"
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "application/geo+json; charset=utf-8"]
+        }
+
+    it "will not override vendored media types like application/vnd.pgrst.object" $
+      request methodGet "/projects?id=eq.1" (acceptHdrs "application/vnd.pgrst.object") ""
+        `shouldRespondWith`
+        [json|{"id":1,"name":"Windows 7","client_id":1}|]
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "application/vnd.pgrst.object+json; charset=utf-8"]
+        }
+
+  context "matches requested media type correctly" $ do
+    -- https://github.com/PostgREST/postgrest/issues/1462
+    it "will match image/png according to q values" $ do
+      r1 <- request methodGet "/rpc/ret_image" (acceptHdrs "image/png, */*") ""
+      liftIO $ do
+        simpleBody r1 `shouldBe` readFixtureFile "A.png"
+        simpleHeaders r1 `shouldContain` [("Content-Type", "image/png")]
+
+      r2 <- request methodGet "/rpc/ret_image" (acceptHdrs "text/html,application/xhtml+xml,application/xml;q=0.9,image/png,*/*;q=0.8") ""
+      liftIO $ do
+        simpleBody r2 `shouldBe` readFixtureFile "A.png"
+        simpleHeaders r2 `shouldContain` [("Content-Type", "image/png")]
```

Comment:
Clears the case in https://github.com/PostgREST/postgrest/issues/1462, replace `image/png` for `image/webp`. This is more about correct content negotiation.